### ISSUE TITLE
[Question] - "Test timeout of 30000ms exceeded while tearing down "context"."

### ISSUE THREAD

**Issue by imartinflores** [microsoft/playwright#17869]
Hi! @aslushnikov sorry to bother,  Just linked to this issue [https://github.com/microsoft/playwright/issues/17670](https://github.com/microsoft/playwright/issues/17670) - I could see it stopped happening when using less workers. I was using 8 and setting only 2 or 4 the tests pass constantly. 

Is there any where I can look at ? Could that be due to the application not behaving properly when running multiple tests at same time or could be a bug ?

I know it's hard as I Can't share the repo, but the tests are pretty simple and can share info (I hope it helps), if it does not help at all, please feel free to just close it, no problem at all!:

[playwright.txt](https://github.com/microsoft/playwright/files/9723447/playwright.txt)

And the config I am using:

```js
projects:[
  {
     name: 'chromium',
     use: {
        headless: true,
        ...devices['Desktop Chrome'],
        launchOptions: {
             proxy: {
                server: myserver,
                username: myuser,
                password: mypassword
             }
        }
     }
  }
]
```
I triggered them using DEBUG=pw:test but I could not see any issues in the logs:

```
pw:test test finished "@MODALERROR @COMPONENT Should show duplicate error" +10s
…um] › components\modal\modalError.spec.js:102:5 › Modal Error Popup › @MODALERROR @COMPONENT Should show duplicate error  
  pw:test teardown page +0ms
  pw:test teardown context +0ms
  pw:test teardown _contextFactory +0ms

  4) [chromium] › components\modal\modalError.spec.js:102:5 › Modal Error Popup › @MODALERROR @COMPONENT Should show duplicate error  
  
Test timeout of 30000ms exceeded while tearing down "context".
```

And with DEBUG=pw:browser:

```
  pw:browser writing frame 1665048293.218038 +48ms
  pw:browser writing frame 1665048293.237627 +57ms
  pw:browser writing frame 1665048293.257545 +18ms
  pw:browser writing frame 1665048293.273618 +29ms
  3) [chromium] › components\modal\modalError.spec.js:59:5 › Bank Modal Error Popup › @MODALERROR @COMPONENT Should show redirect to forgot password page

  Test timeout of 30000ms exceeded while tearing down "context".
```
    
    Thanks!

**Comment by yury-s** [comment]
> Hi! @aslushnikov sorry to bother, Just linked to this issue #17670 - I could see it stopped happening when using less workers. I was using 8 and setting only 2 or 4 the tests pass constantly.
> 

It sounds like a resource problem, you can try increasing timeout to allow more time for the teardown. Usually you don't need to set number of workers manually, playwright should automatically pick optimal value for you. What's the hardware configuration you run this on (how many cores does the CPU have and how much RAM)?

> Is there any where I can look at ? Could that be due to the application not behaving properly when running multiple tests at same time or could be a bug ?

It seems to be just a resource shortage as you probably try to run more tests in parallel than the machine can process with the configured timeout.

**Comment by imartinflores** [comment]
@yury-s Thanks a lot Yury! I did not know Playwright picks the optimal, that is good to know, thank you. Let me try and gather that information for you. But I agree, it could be the App + Resource as I have tried a different Test Suite Today, and noticed some flaky tests when running parallel jobs in Azure. When I ran only 1 job in the agents, everything passed. If you want to close it if it is not related to Playwright itself, please do or let me know and I will do it. I can leave a comment with hardware configuration even if the issue is already closed.
Thanks again!

**Comment by yury-s** [comment]
Ok, thanks, I'll close it for now. Feel free to file a new issue if you need further help!

**Comment by meatnordrink** [comment]
Since I found this while trying to debug this error: It can also be triggered by doing something simple like forgetting an `await` before a `toBeVisible` assertion, in which case it can be a bit of a confusing message.

**Comment by TomasHubelbauer** [comment]
Oh my God, yes! This happens when you forget to `await` a Playwright API. Would be awesome to have this surfaced as a more specific error!

**Comment by popooladayo** [comment]
Test finished within timeout of 30000ms, but tearing down "context" ran out of time.
    Please allow more time for the test, since teardown is attributed towards the test timeout budget.
